### PR TITLE
Add helper methods to create mappings

### PR DESCRIPTION
**Pull request by muuki88**
- directory: create mappings from a baseDirectory including the baseDirectory
- contentOf: create mappings from a baseDirectory excluding the baseDirectory

I put this on the `Path` object. Not sure if this is a good place.
The test aren't property based. I probably need to add some corner cases like

- [x] empty `baseDirectory`
- [x] if `baseDirectory` is a file
- [x] non-existing `baseDirectory`

### CODE REVIEW

File: io/src/main/scala/sbt/io/Path.scala [hunk 1]
```diff
@@ -85,6 +85,60 @@ object Path extends Mapper {
...
+   * {{{
+   * File(extras/file1) -> "file1"
+   * File(extras/file2) -> "file2"
+   * ...
+   * }}}
+   *
+   * @example Add a static directory "extra" and re-map the destination to a different path
+   * {{{
+   * mappings in Universal ++= contentOf(baseDirectory.value / "extra").map {
+   *   case (src, destination => src -> s"new/path/$destination"
```
Comment (dwijnand) [inline io/src/main/scala/sbt/io/Path.scala]: this example doesn't compile (missing `)`) and is missing an explanation at the end.

File: io/src/main/scala/sbt/io/Path.scala [hunk 2]
```diff
@@ -85,6 +85,60 @@ object Path extends Mapper {
...
+   * mappings in Universal ++= directory(baseDirectory.value / "extra")
+   * }}}
+   *
+   * The resulting mappings sequence will look something like this
+   *
+   * {{{
+   * File(extras/file1) -> "extras/file1"
+   * File(extras/file2) -> "extras/"file2"
+   * ...
+   * }}}
```
Comment (dwijnand) [inline io/src/main/scala/sbt/io/Path.scala]: This example seems to be wrong. I'm getting:

```scala
scala> for ((file, path) <- Path directory (file("base") / "extra")) println(s"""$file -> "$path"""")
base/extra -> "extra"
base/extra/file1 -> "extra/file1"
base/extra/file2 -> "extra/file2"
```

File: io/src/main/scala/sbt/io/Path.scala [hunk 3]
```diff
@@ -85,6 +85,60 @@ object Path extends Mapper {
...
+   * mappings in Universal ++= contentOf(baseDirectory.value / "extra")
+   * }}}
+   *
+   * The resulting mappings sequence will look something like this
+   *
+   * {{{
+   * File(extras/file1) -> "file1"
+   * File(extras/file2) -> "file2"
+   * ...
+   * }}}
```
Comment (dwijnand) [inline io/src/main/scala/sbt/io/Path.scala]: This example seems to be wrong. I'm getting:

```scala
scala> for ((file, path) <- Path contentOf (file("base") / "extra")) println(s"""$file -> "$path"""")
base/extra/file1 -> "file1"
base/extra/file2 -> "file2"
```

**Review by dwijnand — CHANGES_REQUESTED** [review]
I've a few comments, but also I think this should probably live with its brothers in `Mapper` (PathMapper.scala).

Thank you for taking the time to PR!

File: io/src/main/scala/sbt/io/PathMapper.scala [hunk 4]
```diff
@@ -111,6 +111,60 @@ abstract class Mapper {
...
+   * mappings ++= directory(baseDirectory.value / "extra")
+   * }}}
+   *
+   * The resulting mappings sequence will look something like this
+   *
+   * {{{
+   * File($baseDirectory/extras/file1) -> "extras/file1"
+   * File($baseDirectory/extras/file2) -> "extras/"file2"
+   * ...
+   * }}}
```
Comment (dwijnand) [inline io/src/main/scala/sbt/io/PathMapper.scala]: Isn't this still missing `$baseDirectory/extra -> "extra"`?

File: io/src/main/scala/sbt/io/PathMapper.scala [hunk 5]
```diff
@@ -111,6 +111,60 @@ abstract class Mapper {
...
+   * @example In order to create mappings for a static directory "extra" add
+   * {{{
+   * mappings ++= directory(baseDirectory.value / "extra")
+   * }}}
+   *
+   * The resulting mappings sequence will look something like this
+   *
+   * {{{
+   * File($baseDirectory/extras/file1) -> "extras/file1"
+   * File($baseDirectory/extras/file2) -> "extras/"file2"
```
Comment (dwijnand) [inline io/src/main/scala/sbt/io/PathMapper.scala]: there's an extra `"`

File: io/src/main/scala/sbt/io/PathMapper.scala [hunk 6]
```diff
@@ -111,6 +111,60 @@ abstract class Mapper {
...
+   * {{{
+   * File($baseDirectory/extras/file1) -> "file1"
+   * File($baseDirectory/extras/file2) -> "file2"
+   * ...
+   * }}}
+   *
+   * @example Add a static directory "extra" and re-map the destination to a different path
+   * {{{
+   * mappings ++= contentOf(baseDirectory.value / "extra").map {
+   *   case (src, destination => src -> s"new/path/$destination"
```
Comment (dwijnand) [inline io/src/main/scala/sbt/io/PathMapper.scala]: there's still a missing `)` here

**Comment by eed3si9n** [comment]
@dwijnand What's the status on this PR?

**Comment by dwijnand** [comment]
I just submitted the review that I had pending from 2 weeks ago..

Haven't reviewed the latest 2 commits.

**Comment by muuki88** [comment]
Sorry for the long standing PR. Finally got time to clean this up

**Review by eed3si9n — APPROVED** [review]

**Review by dwijnand — APPROVED** [review]
Thanks Muki!
